include/fmt/compile.h

```diff
@@ -37,6 +37,28 @@ struct is_compiled_string : std::is_base_of<compiled_string, S> {};
  */
 #define FMT_COMPILE(s) FMT_STRING_IMPL(s, fmt::detail::compiled_string)
 
+#if defined(__cpp_nontype_template_parameter_class) && \
+    (!FMT_GCC_VERSION || FMT_GCC_VERSION >= 903)
```

Comment:
I suggest moving this into a separate macro `FMT_USE_NONTYPE_TEMPLATE_PARAMETERS` since it is used more than once (and to make it controllable).

Reply:
Done